docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,132 @@
+# APIMan Integration
+
+## Abstract
+This proposes a design for integrating [APIMan](http://www.apiman.io/) with OpenShift in order to provide  micro-services governance for API service providers.
+
+## Motivation
+Providers of services may have the need to control who and how their services are consumed.  This may be realized as policies defined by providers to control a service
+in any number of ways (e.g security, throttling/quota, billing and metrics)<sup>[1](#r1)</sup>. 
+
+## Constraints and Assumptions
+### Authentication
+The OAuth token identifying a user in OpenShift will be utilized by APIMan to associate service policy to their identity.  This will ensure a user's identity is 
+ubiquitous across the cluster when interacting with OpenShift and the APIMan management console.
+
+### Deployment
+#### Communication
+APIMan Components (e.g. gateway, management interface) integrated with the cluster will utilize mutual TLS for internal communication.
+
+#### Storage
+APIMan is capable of using several backends (e.g. ElasticSearch, MySql) to store configuration, policy, and metrics.  The initial integration will utilize ElasticSearch with the goal of reusing the same ElasticSearch cluster that is deployed for aggregated logging.  It is assumed when the backend resides on the cluster, [Origin Aggregated logging](https://github.com/openshift/origin-aggregated-logging)<sup>[4](#r4)</sup> will be deployed before APIMan. 
+The [ACL plugin](https://github.com/fabric8io/openshift-elasticsearch-plugin)<sup>[2](#r2)</sup> that is deployed
+as part of aggregated logging will be updated to allow access from the gateway. Additionally, the ElasicSearch index management<sup>[3](#r3)</sup>  will be modified so that it will not cull APIMan data.
+
+
+Cluster administrators can configure APIMan to use an alternate, off cluster ElasticSearch instance if desired.  Some organizations may have an existing instance of ElasticSearch cluster that already defines
+policy, and adminstrators may desire to use it instead of the one provided with aggregated logging.
+
+## Use Cases
+* **UC01** As an API provider, I want to navigate from the OpenShift web console to the policy management interface, so I can define my service policies.
+* **UC02** As an API provider, I want to explicitly expose my service when navigating to the policy managment interface, so that it is consumable according to my management policy.  
+* **UC03** As a cluster administrator, I want a gateway to route API traffic only, so that it manages traffic based on a service's policy.
+* **UC04** As a cluster administrator, I want to deploy APIMan and its components reusing existing infrastructure components where possible (e.g. ElasticSearch), so I can minimize infrastructure components.
+* **UC05** As a cluster administrator, I want to reuse the aggregated logging CA when generating certs for the APIMan components, so I can reuse the ElasticSearch cluster.
```

Comment:
We can't - the CA key is thrown away after generating the certs. Instead, we think we can add a new throwaway CA each time we need new certs and add that CA to the CA truststore in the secret.